server/src/Service/ConnectionService.php

```diff
@@ -0,0 +1,78 @@
+<?php
+
+namespace App\Service;
+
+use Doctrine\DBAL\Connection;
+use Doctrine\DBAL\DriverManager;
+
+class ConnectionService implements ConnectionServiceInterface {
+
+	/**
+	 * @var array
+	 */
+	private $config;
+
+	/**
+	 * @var Connection
+	 */
+	private $conn;
+
+	/**
+	 * @param array $config
+	 */
+	public function __construct( $config ) {
+		$this->config = $config;
+	}
+
+	/**
+	 * @return Connection
+	 * @throws \Exception
+	 */
+	public function getConnection() {
+		if ( !$this->conn ) {
+			throw new \Exception( 'no connection' );
+		}
+
+		return $this->conn;
+	}
+
+	/**
+	 * @param string $wiki
+	 * @return Connection
+	 */
+	public function connect( $wiki ) {
+		if ( isset( $this->conn ) ) {
+			$conn = $this->conn;
+		} else {
```

Comment:
nit pick... why not do `if ( ! isset( $this->conn ) )` and always return `$this->conn` at the bottom and skip the above completely?